lib/transaction/priority_queue.go

```diff
@@ -134,11 +135,36 @@ func (spq *PriorityQueue) Push(txn *ValidTransaction) (common.Hash, error) {
 	spq.currOrder++
 	heap.Push(&spq.pq, item)
 	spq.txs[hash] = item
+	if spq.nextPushWatcher != nil {
+		close(spq.nextPushWatcher)
+	}
```

Comment:
This code can cause a panic if `close(spq.nextPushWatcher)` is called more than once, I notice that you only assign a value to `spq.nextPushWatcher` when the method  `NextPushWatcher` is called. My question is: Let's say we call `NextPushWatcher` and then the `Push` method is called closing the channel how do we ensure that when the `Push` method is called a second time the `close(spq.nextPushWatcher)` not happen?

Reply:
Sorry, miss clicking on the approval 😞

Reply:
Indeed, I guess we could check if the channel is closed

```go
var closed bool
select {
case _, closed = <- spq.nextPushWatcher:
default:
}

if !closed {
  close(spq.nextPushWatcher)
}
```

Or instead, perhaps cleaner, change it to create a new signal channel for each call to `NextPushWatcher`, add each new channel to a slice in the queue struct, and close each in a for loop in the `Push` method (and clear the slice of channels). That would work I think right? 🤔